File: lib/src/model/restore_object_request.dart
import 'generic_request.dart';
import 'restore_configuration.dart';

class RestoreObjectRequest extends GenericRequest {
  RestoreObjectRequest(String bucketName, String key,
      [this.restoreConfiguration])
      : super(bucketName: bucketName, key: key);

  RestoreConfiguration? restoreConfiguration;
}

File: lib/src/model/RestoreObjectRequest.java
package com.alibaba.sdk.android.oss.model;

 class RestoreObjectRequest extends OSSRequest {
     String bucketName;
     String objectKey;

     String getBucketName() {
        return bucketName;
    }

     void setBucketName(String bucketName) {
        this.bucketName = bucketName;
    }

     String getObjectKey() {
        return objectKey;
    }

     void setObjectKey(String objectKey) {
        this.objectKey = objectKey;
    }
}
